alc/backends/wasapi.cpp

```diff
@@ -1071,11 +1076,24 @@ HRESULT WasapiPlayback::openProxy(const char *name)
         mDevice->DeviceName = std::string{DevNameHead} + name;
     else
         mDevice->DeviceName = DevNameHead + DeviceHelper::get_device_name_and_guid(mMMDev).first;
+
+#if defined(ALSOFT_UWP)
+    mDefaultChangeHandler = MediaDevice::DefaultAudioRenderDeviceChanged +=
+        ref new TypedEventHandler<Platform::Object ^, DefaultAudioRenderDeviceChangedEventArgs ^>(
+            [this](Platform::Object ^ sender, DefaultAudioRenderDeviceChangedEventArgs ^ args) {
+        if (args->Role == AudioDeviceRole::Default)
+            mDevice->handleDisconnect("Default audio device changed %s", wstr_to_utf8(args->Id->Data()).c_str());
```

Comment:
I don't think the device should be set to disconnect just because the default changed. As long as the current device is still usable, it should keep working. There app could have a reason for using the device it does and not want to change if the default does, after all.

I do intend to add an API to notify the app when the default device changes (along with possibly device add/remove notifications), which would allow the app to more smoothly change outputs as it wants.

Reply:
> I don't think the device should be set to disconnect just because the default changed. As long as the current device is still usable, it should keep working. There app could have a reason for using the device it does and not want to change if the default does, after all.
> 
> I do intend to add an API to notify the app when the default device changes (along with possibly device add/remove notifications), which would allow the app to more smoothly change outputs as it wants.

I have add new event: `AL_EVENT_TYPE_DEFAULT_DEVICE_CHANGED_SOFT`, please help review